src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/ApplicationIconValueProvider.cs

```diff
@@ -73,8 +73,32 @@ internal sealed class ApplicationIconValueProvider : InterceptingPropertyValuePr
                 propertyValue = PathHelper.MakeRelative(_unconfiguredProject, unevaluatedPropertyValue);
             }
 
-            await _sourceItemsProvider.AddAsync(Content.SchemaName, propertyValue);
+            string existingPropertyValue = await defaultProperties.GetEvaluatedPropertyValueAsync(ConfigurationGeneral.ApplicationIconProperty);
+            IProjectItem? existingItem = await GetExistingContentItemAsync(existingPropertyValue);
+            if (existingItem != null)
+            {
+                await existingItem.SetUnevaluatedIncludeAsync(propertyValue);
+            }
+            else
+            {
+                await _sourceItemsProvider.AddAsync(Content.SchemaName, propertyValue);
+            }
+
             return propertyValue;
         }
+
+        private async Task<IProjectItem?> GetExistingContentItemAsync(string existingPropertyValue)
+        {
+            foreach (IProjectItem item in await _sourceItemsProvider.GetItemsAsync(Content.SchemaName))
```

Comment:
ISourceItemProvider has two methods that handle this for you; FindItemByNameAsync and GetItemAsync that takes an evaluated include.

Reply:
This is a simplified version of something I did in a previous PR:
https://github.com/dotnet/project-system/blob/76e03d472b359876e5fbacb452edc3d6ca97d675/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/PackagePropertyPage/PackageIconValueProvider.cs#L60-L76

There is a `GetItemsAsync` that takes an item type and evaluated include, but not a `GetItemAsync`, so it will be very similar to what is there. And `FindItemByNameAsync` has no filter for item type, so I'd still have to check that anyway. I plan on making some shared code from my previous PRs that did similar things. But, not consolidating the code in this PR.

Reply:
Can you test the following things to make sure this handles it correctly:

1) An item outside of the project file in an imports.
2) An item included as part of a glob (*.*)
3) An ApplicationIconProperty with a variable in it $(FooBar)
4) An ApplicationIconProperty with bad path characters "C:\$$$".

Reply:
I've added `item.PropertiesContext.IsProjectFile` check to this so that scenarios 1 and 2 are not an issue. This exists in the code I linked above, but not in the method itself (as part of caller logic). For 3 and 4, it just reverts to the previous value (if-statement inside `OnSetPropertyValueAsync` code). I didn't support variables in the setter because it basically allows anything to be provided... which invalidates the purpose of having logic for the setter. Unless... there is a way to process an arbitrary value (like this unevaluated property) and evaluate it during the setter; even though it isn't written to the project yet?